ext_airflow/wrapper.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import annotations
+
+import os
+import pkgutil
+import subprocess
+import sys
+from pathlib import Path
+
+import structlog
+
+from meltano_extension_sdk import models
+from meltano_extension_sdk.extension import ExtensionBase
+from meltano_extension_sdk.process import Invoker, log_subprocess_error
+
+log = structlog.get_logger("airflow_extension")
+
+
+class Airflow(ExtensionBase):
+    def __init__(self):
+
+        self.app_name = "airflow_extension"
+        self.airflow_bin = "airflow"
+        self.airflow_invoker = Invoker(self.airflow_bin, env=os.environ.copy())
+
+        self.airflow_home = os.environ.get("AIRFLOW_HOME") or os.environ.get(
+            f"{self.app_name}_AIRFLOW_HOME"
+        )
+        if not self.airflow_home:
+            log.debug("env dump", env=os.environ)
+            log.error(
+                "AIRFLOW_HOME not found in environment, unable to function without it"
+            )
+            sys.exit(1)
+
+        self.airflow_cfg_path = Path(
+            os.environ.get("AIRFLOW_CONFIG", f"{self.airflow_home}/airflow.cfg")
+        )
+        self.airflow_core_dags_path = Path(
+            os.path.expandvars(
+                os.environ.get(
+                    "AIRFLOW__CORE__DAGS_FOLDER",
+                    f"{self.airflow_home}/orchestrate/dags",
+                )
+            )
+        )
+        # Configure the env to make airflow installable without GPL deps.
+        os.environ["SLUGIFY_USES_TEXT_UNIDECODE"] = "yes"
+
+    def pre_invoke(self):
+        self._create_config()
+        self._initdb()
+
+    def initialize(self, force: bool = False):
```

Comment:
@pandemicsyn - should this also create `airflow.cfg` and/or `webserver_config.py`?

Reply:
I went that route in Superset and so superset doesn't use the pre_invoke pattern. I'd prefer Airflow to do same and use initialize(). During one of the call's we chatted about having one plugin for each pattern, but would definitely be ok moving away from that.